src/common/internal/logging/log_message.ts

```diff
@@ -1,26 +1,43 @@
 import { ErrorWithExtra } from '../../util/util.js';
 import { extractImportantStackTrace } from '../stack.js';
 
+import { LogMessageRawData } from './result.js';
+
 export class LogMessageWithStack extends Error {
   readonly extra: unknown;
 
   private stackHiddenMessage: string | undefined = undefined;
 
-  constructor(name: string, ex: Error | ErrorWithExtra) {
-    super(ex.message);
+  /** Wrap an Error (which was created to capture the stack at that point) into a
+   * LogMessageWithStack (which has extra stuff for good log messages).
+   */
+  static wrapError(name: string, ex: Error | ErrorWithExtra) {
+    return new LogMessageWithStack({
+      name,
+      message: ex.message,
```

Comment:
Do you want to send `ex.constructor.name`?

Reply:
`name` is sort of meant to override it. Chrome includes the original name in the `stack`, but other browsers don't so we should preserve it somehow for cases where it matters (I think just `rec.threw()`).